### PR TITLE
Include Memory Header for shared_ptr

### LINKED ISSUE #82: shared_ptr is not in std namespace on Linux based systems
## Current behavior
shared_ptr is not in std namespace on Linux based systems and will therefore cause compilation error.

## Proposed
Need to add  `#include <memory>` and will compile.

### PR DESCRIPTION
**Pull request by jdmattern-noaa**
Including memory header `#include <memory>` for use of shared_ptr on Linux-based systems.

Fixes #82 

## Additions
Added header to several files

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

## Testing checklist (automated report can be put here)

1.

### Target Environment support

- [ ] Linux

### CODE REVIEW

File: src/models/tshirt/Tshirt.cpp [hunk 1]
```diff
@@ -121,4 +121,4 @@ namespace tshirt {
         return 0;
     }
 
-}
\ No newline at end of file
+}
```
Comment (hellkite500) [inline src/models/tshirt/Tshirt.cpp:124]: Why is this a "change"?

Comment (jdmattern-noaa) [reply src/models/tshirt/Tshirt.cpp:124]: I am not sure how that got in there actually. Is it not part of the namespace from the top? I wasn't able to finish compiling this due to the make_unique from 4.8.5.

Comment (jdmattern-noaa) [reply src/models/tshirt/Tshirt.cpp:124]: Oh I think this just means no newline at end of file. I think it must be a gedit thing from opening the file.

File: models/kernels/Nonlinear_Reservoir.hpp [hunk 2]
```diff
@@ -365,4 +366,4 @@ class Nonlinear_Reservoir
     outlet_vector_type outlets;
 };
 
-#endif  // NONLINEAR_RESERVOIR_HPP
\ No newline at end of file
+#endif  // NONLINEAR_RESERVOIR_HPP
```
Comment (jdmattern-noaa) [inline models/kernels/Nonlinear_Reservoir.hpp:369]: Same thing here about the no newline with gedit.

**Pull request merged**